### ISSUE TITLE
Add Mechanism class

### ISSUE THREAD

**Issue by aidnem** [team401/coppercore#35]
**Purpose**
Create a Mechanism class to separate logic for individual mechanisms and abstract it out of the subsystems without causing duplicated code in the IOs.

**Scope**
- [ ] Create a mechanism class that handles an IO and updates inputs and applies outputs
- [ ] Create an IO interface that other IOs can implement, so that the mechanism class can handle IOs automatically

Based on: [this discussion](https://github.com/team401/coppercore/discussions/33#discussioncomment-11194379).

**Comment by aidnem** [comment]
This is actually very sketchy to implement with generics, we should reconsider if we need to explicitly add mechanisms to coppercore tonight at shop.

**Comment by avidraccoon** [comment]
@aidnem Is this still planned?

**Comment by aidnem** [comment]
No, we realized we don't really need it. You can just implement a mechanism yourself.
